src/app/public/modules/phone-field/phone-field.component.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  OnDestroy,
+  OnInit,
+  Output,
+  Input
+} from '@angular/core';
+
+require('intl-tel-input/build/js/utils');
+
+require('intl-tel-input/build/js/intlTelInput');
+
+/**
+ * NOTE: We can not type these due the the current @types/intl-tel-input version having an
+ * undeclared type which causes linting errors.
+ */
+declare var intlTelInputUtils: any;
+declare var intlTelInputGlobals: any;
+
+import {
+  SkyPhoneFieldCountry
+} from './types';
+
+@Component({
+  selector: 'sky-phone-field',
+  templateUrl: './phone-field.component.html',
+  styleUrls: ['./phone-field.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyPhoneFieldComponent implements OnDestroy, OnInit {
+
+  @Input()
+  public set defaultCountry(value: string) {
+    if (value !== this._defaultCountry) {
+      this._defaultCountry = value;
+
+      this.defaultCountryData = this.countryData.find(country => country.iso2 === value);
+      this.sortCountriesWithSelectedAndDefault(this.selectedCountry);
+    }
+  }
+
+  public get defaultCountry(): string {
+    return this._defaultCountry;
+  }
+
+  @Output()
+  public selectedCountryChange = new EventEmitter<SkyPhoneFieldCountry>();
+
+  public countryData: SkyPhoneFieldCountry[];
+
+  public disabled = false;
```

Comment:
I looked in the template and this property is only disabling the country selection mechanism. To make this more clear from the directive's perspective (i.e., "What am I disabling?") maybe we could rename this to `countrySelectDisabled`? From the directive, it would look like:

```
@Input()
public set disabled(value: boolean) {
  this.phoneFieldComponent.countrySelectDisabled = value;
  // ...
}
```

Reply:
Done